spec/README.md

```diff
@@ -0,0 +1,172 @@
+* `MevBlock`: Comes from a block proposal where the reward was obtained via an off-chain agreement using tools such as mev-boost, usually coming as the last transaction in the block.
+* `VanilaBlock`: Comes from a block proposal where the reward was sent via the protocol `fee_recipient`.
+* `PayableDonation`: A donation was made to the smoothing pool contract, sent as a normal transaction detected in the payable.
+* `NonPayableDonation`: A donation was made to the smoothing pool contract, sent via a smart contract, which is not seen in the payable.
+
+## Subscribe/unsubscribe/ban
+
+Rewards are only shared among subscribed participants in the pool. Hereunder it's explained the different ways in which a validator can join or leave the pool. Joining can be done with manual or automatatic subscription. And leaving can be done by unsubscribing to the pool or by being banned from it.
+
+**Subscribing** to the pool:
+* `ManualSubscription`: If any validator sends a `MevBlock` or `VanilaBlock` reward to the smoothing pool contract `POOL_CONTRACT_ADDRESS` it is considered automatically subscribed into the pool, and will start accruing rewards from that moment. This type of subscription doesn't require any collateral or lock up of funds, since by successfully proposing a block with the correct fee recipient, we consider that this validator has enough skin in the game. However, since block proposals are a rare event, it can take weeks or even months for a validator to get automatically subscribed. This is not ideal because it won't be leveraging the benefits of the smoothing pool during this time.
+* `AutoSubscription`: On the other hand, a validator can start earning rewards from the very beginning if it adds `COLLATERAL_GWEI` amount as collateral. This collateral can be deposited by calling the register function in the oracle smart contract from any account, but specifying the correct amount and a valid `validatorIndex`. This transaction will be detected by the oracle, and will subscribe the validator, starting to accrue `PendingRewards` just one block after being recognized. This type of subscription allows the validator to start earning rewards without having to wait weeks or months until a proposal is detected. Note that the subscription is only considered valid if `collateral>=COLLATERAL_GWEI`, which its added to the validator `PendingRewards`. This collateral is returned after the first valid block proposal. See `PendingRewards`, `AccumulatedRewards`. Note that any address can subscribe a validator to the pool, but just calling this smart contract function.
```

Comment:
Right now, the subscription is only valid if `collateral==COLLATERAL_GWEI` not if `collateral>=COLLATERAL_GWEI` like the spec says. See https://github.com/dappnode/mev-sp-contracts/blob/d1a7a8894d5cf7e49924287cbacdcb6ac29be142/contracts/DappnodeSmoothingPool.sol#L119